pkgs/by-name/bi/bitwarden-desktop/package.nix

```
@@ -150,33 +167,37 @@ buildNpmPackage rec {
   installPhase = ''
     runHook preInstall
 
-    mkdir $out
-
-    pushd apps/desktop/dist/linux-${lib.optionalString stdenv.hostPlatform.isAarch64 "arm64-"}unpacked
-    mkdir -p $out/opt/Bitwarden
-    cp -r locales resources{,.pak} $out/opt/Bitwarden
-    popd
-
-    makeWrapper '${lib.getExe electron}' "$out/bin/bitwarden" \
-      --add-flags $out/opt/Bitwarden/resources/app.asar \
-      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations --enable-wayland-ime=true}}" \
-      --set-default ELECTRON_IS_DEV 0 \
-      --inherit-argv0
-
-    # Extract the polkit policy file from the multiline string in the source code.
-    # This may break in the future but its better than copy-pasting it manually.
-    mkdir -p $out/share/polkit-1/actions/
-    pushd apps/desktop/src/key-management/biometrics
-    awk '/const polkitPolicy = `/{gsub(/^.*`/, ""); print; str=1; next} str{if (/`;/) str=0; gsub(/`;/, ""); print}' os-biometrics-linux.service.ts > $out/share/polkit-1/actions/com.bitwarden.Bitwarden.policy
-    popd
-
-    pushd apps/desktop/resources/icons
-    for icon in *.png; do
-      dir=$out/share/icons/hicolor/"''${icon%.png}"/apps
-      mkdir -p "$dir"
-      cp "$icon" "$dir"/${icon}.png
-    done
-    popd
+    ${lib.optionalString stdenv.hostPlatform.isDarwin ''
+      mkdir -p $out/Applications
+      cp -r apps/desktop/dist/mac*/Bitwarden.app $out/Applications
+      makeWrapper $out/Applications/Bitwarden.app/Contents/MacOS/Bitwarden $out/bin/bitwarden
+    ''}
+
+    ${lib.optionalString stdenv.hostPlatform.isLinux ''
+      mkdir -p $out/opt/Bitwarden
```

```suggestion
  '' + lib.optionalString stdenv.hostPlatform.isDarwin ''
      mkdir -p $out/Applications
      cp -r apps/desktop/dist/mac*/Bitwarden.app $out/Applications
      makeWrapper $out/Applications/Bitwarden.app/Contents/MacOS/Bitwarden $out/bin/bitwarden
  '' + lib.optionalString stdenv.hostPlatform.isLinux ''
      mkdir -p $out/opt/Bitwarden
```
Please use this style of conditional multiline strings

---

Have we decided that one way is better than the other way?

---

Having the conditions inline requires extra variable interpolation and also makes it super easy to get indentation in the final scripts wrong which is not important here but where it is, it easily bites you into the finger. Also not inlining the strings makes it easier to understand and grasp when reading the code.